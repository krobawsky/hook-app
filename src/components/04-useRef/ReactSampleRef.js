import React, { useState } from 'react';
import '../02-useEffect/effects.css';

import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const ReactSampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            
            <h1>Sample Ref</h1>
            <hr/>
            

            { show && <MultipleCustomHooks /> }

            <button
                className="btn btn-primary mt-4"
                onClick={() => {
                    setShow(!show);
                }}
            >
                Show hide
            </button>

        </div>
    )
}
