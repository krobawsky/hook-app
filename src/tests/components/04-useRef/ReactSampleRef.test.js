import React from 'react';
import { shallow } from 'enzyme';
import { ReactSampleRef } from '../../../components/04-useRef/ReactSampleRef';

describe('Pruebas en <ReactSampleRef />', () => {
    
    const wrapper = shallow(<ReactSampleRef />);

    test('debe mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(false);

    })

    test('debe mostrarse <MultipleCustomHooks />', () => {
        
        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(true);

    })

})