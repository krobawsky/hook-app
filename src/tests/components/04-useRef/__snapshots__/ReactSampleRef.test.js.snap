// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ReactSampleRef /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Sample Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Show hide
  </button>
</div>
`;
